### ISSUE TITLE
Compiling unicode-transforms fails 

### ISSUE THREAD

**Issue by phadej** [ghcjs/ghcjs#703]
```
[ 7 of 14] Compiling Data.Unicode.Properties.Decompositions ( Data/Unicode/Properties/Decompositions.hs, dist/build/Data/Unicode/Properties/Decompositions.js_o )
Ix{Int}.index: Index (1025) out of range ((1,1024))
cabal: Failed to build unicode-transforms-0.3.4
```

https://github.com/harendra-kumar/unicode-transforms/blob/master/Data/Unicode/Properties/Decompositions.hs - I'm not sure if some JS-string weirdness does it here

```javascript
('💩').length
2
```

```haskell
λ> putStrLn "\128169"
💩
λ> length "\128169"
1
```

**Comment by luite** [comment]
Yikes, scary function. Probably running into some hardcoded size limits of the compiler.

Shouldn't be hard to raise.

**Comment by luite** [comment]
fixed in d594a012e59a21dda058ddadb65e5a4e3d696e16

but I had to increase the maximum stack size to make the `unicode-transforms` testsuite link. this suggests that the generated code is probably not very good, and that the linker is doing something wrong too.

**Comment by phadej** [comment]
@luite, thanks
